packages/back-end/src/services/jobLifecycle.ts

```diff
@@ -0,0 +1,51 @@
+import { Job, JobAttributesData } from "agenda";
+import { logger } from "back-end/src/util/logger";
+import { MAX_QUERY_TIMEOUT_MS } from "back-end/src/util/secrets";
+
+const TOUCH_INTERVAL_MS = 3 * 60 * 1000;
+const JOB_TIMEOUT_MS = MAX_QUERY_TIMEOUT_MS + 1000; // Allow some buffer for the query client to close properly after it times out
+
+// As some functions may take a while to run, this will update the job's lock if the process is still running
+export const addJobLifecycleChecks = <T extends JobAttributesData>(
+  fn: (job: Job<T>) => Promise<void>
+) => async (job: Job<T>) => {
+  let touchTimer: NodeJS.Timeout | null = null;
+  let timeoutTimer: NodeJS.Timeout | null = null;
+  let finished = false;
+
+  function startTouch() {
+    touchTimer = setInterval(() => {
+      if (!finished) {
+        job.touch().catch((e) => {
+          logger.error(e, `Failed to touch Agenda job ${job.attrs.name}`);
```

Comment:
These are not the correct arguments for logger.error.  It really only outputs the first arg as a message.  So I think we intend for this to be: 
```
logger.error(`Error while trying to touch Agenda job ${job.attrs.name}: ${e}`);
```